src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerRemoveDocument.java

```diff
@@ -140,4 +147,67 @@ private List<ListValue<ScannedDocument>> findScannedDocumentsForRemoval(final Li
 
         return scannedDocsToRemove;
     }
+
+    private void handleDeletionOfGeneralApplicationDocuments(CaseData beforeCaseData, CaseData currentCaseData) {
+        List<ListValue<DivorceDocument>> uploadedDocsToRemove = new ArrayList<>();
+        List<ListValue<DivorceDocument>> generalAppDocs = new ArrayList<>();
+
+        uploadedDocsToRemove.addAll(findDocumentsForRemoval(
+            beforeCaseData.getDocuments().getDocumentsUploaded(),
+            currentCaseData.getDocuments().getDocumentsUploaded()
+        ));
+
+        if (!uploadedDocsToRemove.isEmpty()) {
+            generalAppDocs = uploadedDocsToRemove
+                .stream()
+                .filter(divorceDocumentListValue ->
+                    divorceDocumentListValue.getValue().getDocumentType().equals(DocumentType.GENERAL_APPLICATION))
+                .collect(Collectors.toList());
+        }
+
+        if (!generalAppDocs.isEmpty()) {
+            for (ListValue<DivorceDocument> generalAppDoc : generalAppDocs) {
+                log.info("General App Doc to remove : {} ", generalAppDoc.getValue().getDocumentLink().getUrl());
+                handleDeletionOfGeneralApplicationDocument(currentCaseData, generalAppDoc.getValue());
+            }
+        }
+    }
+
+    void handleDeletionOfGeneralApplicationDocument(CaseData caseData, DivorceDocument document) {
```

Comment:
I would explicitly make this Private for consistency

Reply:
Made the change